Charts, docs, readability and other changes

khaosdoctor opened the pull request.

khaosdoctor committed:
Fix Typos

khaosdoctor committed:
Merge branch 'start' of https://github.com/arschles/http-add-on into start

khaosdoctor committed:
Implement object deletion

khaosdoctor committed:
Add prefix for logging

khaosdoctor committed:
Remove unused status interface

khaosdoctor committed:
Add status check for "Ready"

khaosdoctor committed:
Golint

khaosdoctor committed:
Generate CRD Manifests

khaosdoctor committed:
Reorder RBAC Roles

khaosdoctor committed:
Change file namings

khaosdoctor committed:
Remove CLI part for now

khaosdoctor committed:
Add internal build context for operator

khaosdoctor committed:
Remove Rust-related things

khaosdoctor committed:
Add comment docs and reorder packages

khaosdoctor committed:
Change variable name and add package docs

khaosdoctor committed:
Add package docs

khaosdoctor committed:
Add package docs

khaosdoctor committed:
Rename var for readability

khaosdoctor committed:
Adds package docs

khaosdoctor committed:
Remove commented code

khaosdoctor committed:
Remove unused admin echo server and rename current one

khaosdoctor committed:
Add more logging and docs

khaosdoctor committed:
Fix doc

khaosdoctor committed:
Update readme on cli

khaosdoctor committed:
Add scaffold to operator chart

khaosdoctor committed:
Add docs

khaosdoctor committed:
Golint

khaosdoctor committed:
Update name in kustomization

khaosdoctor committed:
Add descriptions to fields

khaosdoctor committed:
Update default image name

khaosdoctor committed:
Change resource names

khaosdoctor committed:
Add version tag to kustomization

khaosdoctor committed:
Merge branch 'start' of https://github.com/arschles/http-add-on into start

khaosdoctor committed:
Add make command for charts

khaosdoctor committed:
Create base values file

khaosdoctor committed:
Add patch for image name

khaosdoctor committed:
Add environment variable to manager

tomkerkhove commented on charts/operator/.helmignore:
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Can we move this to `keda-http-operator` please?

khaosdoctor replied:
Sure! I'll change it!

khaosdoctor replied:
Done

tomkerkhove commented on charts/operator/Chart.yaml:
```diff
@@ -0,0 +1,23 @@
+apiVersion: v2
+name: operator
```
```suggestion
name: keda-http-operator
```

tomkerkhove commented on charts/operator/Chart.yaml:
```diff
@@ -0,0 +1,23 @@
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 1.16.0
```
This feels more suitable maybe?
```suggestion
appVersion: 0.1.0
```

khaosdoctor replied:
Agree, I was trying to figure out where did helm took this version number

khaosdoctor committed:
Update charts/operator/Chart.yaml

Co-authored-by: Tom Kerkhove <kerkhove.tom@gmail.com>

khaosdoctor committed:
Update charts/operator/Chart.yaml

Co-authored-by: Tom Kerkhove <kerkhove.tom@gmail.com>

khaosdoctor committed:
Correct dir name and reuse step

khaosdoctor committed:
Rename dir to keda-http-operator

khaosdoctor committed:
Add envs to deployment pkg

khaosdoctor committed:
Add new statuses for extended objetct

khaosdoctor committed:
Fixing interceptor listening port

khaosdoctor committed:
Include new "Terminating" status

khaosdoctor committed:
Add namespaces for logs

khaosdoctor committed:
Change variable names

khaosdoctor committed:
Add new statuses

khaosdoctor committed:
Code reorder

khaosdoctor committed:
Create interceptor

khaosdoctor committed:
Add missing return

khaosdoctor committed:
Remove comment

khaosdoctor committed:
Fix link

khaosdoctor committed:
Remove all references to external scaler address

This is because we'll use one external scaler for each new application, so there's no point in having a single environment variable controlling the
address of the external scaler.

The address will be defined by the operator for each new app and a new scaler will be created.

khaosdoctor committed:
Fixing external scaler port

khaosdoctor committed:
Code reorder and creation of external scaler

khaosdoctor committed:
Improve code readability

khaosdoctor committed:
Golint

khaosdoctor committed:
Add colons

khaosdoctor committed:
Further clean the code

khaosdoctor committed:
Move functions to other file

khaosdoctor committed:
Rename files

khaosdoctor committed:
Add more meaninful names

khaosdoctor committed:
Rename method

khaosdoctor committed:
Update roles and enums

khaosdoctor committed:
Make yamls

khaosdoctor commented:
Added a few more changes:

- Create the interceptor and the external scaler for each new app (this can be changed later according to what @arschles told me about having a single external scaler)
- Add removal steps for newly created objects
- Removed environment variables that were not needed anymore
- Hardcoded exposed ports so we don't deal with complex port settings
- Added ingresses to the role because we'll need to create them
- Added a new status of "Terminating"
- Reordered the code and renamed files for readability

khaosdoctor committed:
Add next TODO

khaosdoctor committed:
Add todo

arschles commented:
@khaosdoctor I'm gonna merge this now, and here are 3 follow-up issues that I'm going to start working on: https://github.com/kedacore/http-add-on/issues/5, https://github.com/kedacore/http-add-on/issues/4, https://github.com/kedacore/http-add-on/issues/3
